frontend/src/app/[locale]/auth/_components/DatePicker.tsx

```diff
@@ -0,0 +1,165 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import {
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from "@/components/ui/form";
+import {
+	Popover,
+	PopoverContent,
+	PopoverTrigger,
+} from "@/components/ui/popover";
+import {
+	Select,
+	SelectContent,
+	SelectItem,
+	SelectTrigger,
+	SelectValue,
+} from "@/components/ui/select";
+import { cn } from "@/lib/utils";
+import { format, getMonth, getYear, setMonth, setYear } from "date-fns";
+import { Calendar as CalendarIcon } from "lucide-react";
+import { useLocale, useTranslations } from "next-intl";
+import * as React from "react";
+import { useFormContext } from "react-hook-form";
+
+// Import the necessary locales from date-fns
+import { enUS, pl } from "date-fns/locale";
+
+interface DatePickerProps {
+	fromDate: Date;
+	toDate: Date;
+}
+
+export function DatePicker({ fromDate, toDate }: DatePickerProps) {
+	const [date, setDate] = React.useState<Date>(toDate);
+
+	const t = useTranslations("SignUpForm.Calendar");
+
+	const locale = useLocale();
+
+	const calendarLocale = locale === "pl" ? pl : enUS;
+
+	const months = [
```

Comment:
is there a way to maybe not do that? create SignUpForm.Calendar.Months and iterate through all elements?

Reply:
no.